Sources/NIOTransportServices/NIOTSConnectionChannel.swift

```diff
@@ -893,4 +893,53 @@ extension NIOTSConnectionChannel {
         return SynchronousOptions(channel: self)
     }
 }
+
+
+public struct NIOTSConnectionNotInitialized: Error, Hashable {
+    public init() {}
+}
+
+public struct NIOTSChannelIsNotATransportServiceChannel: Error, Hashable {
+    public init() {}
+}
+
+@available(OSX 10.14, iOS 12.0, tvOS 12.0, watchOS 6.0, *)
+extension NIOTSConnectionChannel {
+    fileprivate func metadata(definition: NWProtocolDefinition) throws -> NWProtocolMetadata? {
+        guard let nwConnection = nwConnection else {
+            throw NIOTSConnectionNotInitialized()
+        }
+        return nwConnection.metadata(definition: definition)
+    }
+}
+
+@available(OSX 10.14, iOS 12.0, tvOS 12.0, watchOS 6.0, *)
+extension Channel {
+    /// Retrieves the metadata for a specific protocol from the underlying ``NWConnection``
+    /// - Throws: If `self` isn't a `NIOTS` channel with a `NWConnection` this method will throw
+    /// ``NIOTSChannelIsNotATransportServicesChannel`` or ``NIOTSConnectionNotInitialized``.
+    public func getMetadata(definition: NWProtocolDefinition) -> EventLoopFuture<NWProtocolMetadata?> {
+        guard let channel = self as? NIOTSConnectionChannel else {
+            return self.eventLoop.makeFailedFuture(NIOTSChannelIsNotATransportServiceChannel())
+        }
+        return self.eventLoop.submit { try channel.metadata(definition: definition) }
```

Comment:
Can we add an optional fast-path here?

Reply:
Added in https://github.com/apple/swift-nio-transport-services/pull/163/commits/c6bd6ed8c605d384e1bb43f0ba50468530ea4b1d